lib/dialogue-agent/dialogue-loop.ts

```diff
@@ -655,10 +665,29 @@ export default class DialogueLoop {
         await this.setExpected(null);
     }
 
-    private async _loop(showWelcome : boolean) {
-        // if we want to show the welcome message, we run the policy on the `null` state, which will return the sys_greet intent
-        if (showWelcome)
+    private async _loop(showWelcome : boolean, initialState : string|null) {
+        if (initialState !== null) {
+            if (initialState === 'null') {
+                this._dialogueState = null;
+                await this.setExpected(null);
```

Comment:
in what case this is a string 'null'? should we show welcome in this case?

Reply:
If initialState is set, it means we're resuming an existing conversation, and we should not show the welcome again. initialState will be the exact string `null` if the agent is not asking a question and the state is null (all commands completed).